crates/wasmtime/src/component/func.rs

```diff
@@ -308,11 +314,14 @@ impl Func {
                     self.store_args(store, &options, &params, args, dst)
                 } else {
                     dst.write([ValRaw::u64(0); MAX_FLAT_PARAMS]);
-                    let dst = unsafe {
+
+                    let dst = &mut unsafe {
                         mem::transmute::<_, &mut [MaybeUninit<ValRaw>; MAX_FLAT_PARAMS]>(dst)
-                    };
+                    }
+                    .iter_mut();
+
                     args.iter()
-                        .try_for_each(|arg| arg.lower(store, &options, &mut dst.iter_mut()))
+                        .try_for_each(|arg| arg.lower(store, &options, dst))
```

Comment:
Was this for correctnes or a refactoring? Fine either way of course, just curious for my own edification

Reply:
Correctness.  The original code meant that each argument would overwrite any that came before it in the list.  This was the first bug the fuzz tests found.

Reply:
Aah yes indeed I see the issue now!